pkg/routeagent_driver/handlers/kubeproxy/iptables_iface.go

```diff
@@ -83,38 +83,47 @@ func (kp *SyncHandler) createIPTableChains() error {
 
 func (kp *SyncHandler) updateIptableRulesForInterClusterTraffic(inputCidrBlocks []string, operation Operation) {
 	for _, inputCidrBlock := range inputCidrBlocks {
-		if operation == Add {
-			err := kp.programIptableRulesForInterClusterTraffic(inputCidrBlock)
-			if err != nil {
-				klog.Errorf("Failed to program iptable rule. %v", err)
-			}
-		} else if operation == Delete {
-			// TODO: Handle this use-case
-			klog.Warning("Handle the delete use-case")
+		err := kp.programIptableRulesForInterClusterTraffic(inputCidrBlock, operation)
```

Comment:
Why not retain the `if operation` and write a new method for `cleanupIpTableRulesforInterClusterTraffic`. It will simplify code and make it more readable. Other than ipt.New() nothing else is common between Add/Delete cases.

Reply:
No, its not just ipt.New(), the ruleSpec and the flow will be more or less the same. 
In fact, I tried that initially, but linter was complaining that code is duplicated. So modified it to make linter happy.

Reply:
I checked it locally and it fixed the issue
Maybe we can keep the if operation and just set function pointer to ipt.AppendUnique or ipt.Delete

Reply:
In second thought, if we don't plan to add new logic to intercluster ipt rules,  I think the suggested change is OK

Reply:
Good to know that you did try that way initially.